### ISSUE TITLE
Migrate scheduler to structured logging

### ISSUE THREAD

**Issue by pchan** [kubernetes/kubernetes#105841]
### What would you like to be added?

Enhance scheduler logging and address current issues.

1. Migrate the current unstructured logs to structured logging. 
    - Klog APIs of unstructured logs are found in 2 directories (cmd/kube-scheduler, pkg/scheduler)
    - Primarily files in pkg/scheduler/framework and pkg/scheduler/internal needs to be migrated. I will update this issue with the API breakdown at the file level.
2. One improvement that would be nice to have is logging a unique key for each pod through its scheduling cycle. This was raised in #91633. Need tracing for this, will need to work with SIG instrumentation and do more analysis.

**Authors** 

What to include in your PR

Subject: "Migrate to structured logging"
SIGs: instrumentation, node
Priority: important-soon
Kind: cleanup
HOW TO: [Read the structured logging migration guide.](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md)

Please reference this issue for scheduler migrations. 

Before you submit your PR, check to see if someone has already migrated that component: https://github.com/kubernetes/kubernetes/pulls?q=is%3Apr+is%3Aopen+structured+log+label%3Asig%2Fnode

**Reviewers**: 

Read through the migration guide above. Use the PR query above to find PRs to review. 

### Why is this needed?

To get the benefits of structured logging. 

kubernetes/enhancements#1602

/wg structured-logging 
/sig scheduling
/area logging
/priority important-longterm
/kind cleanup
/cc @kubernetes/wg-structured-logging-reviews
/cc @damemi

**Comment by jyz0309** [comment]
Maybe I can help to migrate the `pkg/scheduler/framework/plugins/volumebinding/binder.go` and  `pkg/scheduler/framework/plugins/volumebinding/assume_cache.go`, can you assign it to me?

**Comment by jyz0309** [comment]
BTW, I think we can divide it into 4 part as @shivanshu1333 said. I have tried to divide it into 4 parts:
<!DOCTYPE html>

  |   |  
-- | -- | --
File | Unstructured log API Calls | Assigned
cmd/kube-scheduler/app/server.go | 2 |  
pkg/scheduler/framework/plugins/nodelabel/node_label.go | 1 |  
pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go | 1 |  
pkg/scheduler/framework/plugins/volumebinding/binder_test.go | 2 |  
pkg/scheduler/framework/plugins/examples/stateful/stateful.go | 1 |  
pkg/scheduler/framework/plugins/noderesources/resource_allocation.go | 2 |  
pkg/scheduler/framework/plugins/serviceaffinity/service_affinity.go | 1 |  
pkg/scheduler/framework/preemption/preemption.go | 1 |  
  |   |  
pkg/scheduler/internal/cache/debugger/dumper.go | 3 |  
pkg/scheduler/internal/cache/debugger/comparer.go | 2 |  
pkg/scheduler/internal/cache/cache.go | 17 |  
pkg/scheduler/internal/cache/node_tree.go | 4 |  
  |   |  
pkg/scheduler/framework/plugins/volumebinding/assume_cache.go | 14 |  
pkg/scheduler/framework/plugins/volumebinding/binder.go | 28 |  
  |   |  
pkg/scheduler/framework/plugins/interpodaffinity/filtering.go | 2 |  
pkg/scheduler/framework/plugins/legacy_registry.go | 4 |  
pkg/scheduler/framework/plugins/nodepreferavoidpods/node_prefer_avoid_pods.go | 1 |  
pkg/scheduler/framework/plugins/nodevolumelimits/csi.go | 1 |  
pkg/scheduler/framework/plugins/podtopologyspread/filtering.go | 4 |  
pkg/scheduler/framework/plugins/volumezone/volume_zone.go | 1 |

**Comment by shivanshu1333** [comment]
Complete migration of scheduler depends on resolution of (([106258 ](https://github.com/kubernetes/kubernetes/issues/106258) or [106276](https://github.com/kubernetes/kubernetes/issues/106276)) and ([106262](https://github.com/kubernetes/kubernetes/issues/106262)))
We still need to migrate `pkg/scheduler/internal/cache/debugger/dumper.go` (2 lines (49 and 60)) and `cmd/kube-scheduler/app/options/configfile.go` (1 line (96)) after resolving the above issues, and update the `hack/.structured_logging` indicating complete migration of scheduler.

**Comment by jyotimahapatra** [comment]
Hi 👋 Im the bug triage shadow. Since we're in code freeze for 1.23, i wanted to check if we can move this issue to milestone 1.24.

**Comment by serathius** [comment]
@jyotimahapatra I think that this issue is just blocked on https://github.com/kubernetes/kubernetes/pull/106305

**Comment by pohly** [comment]
But even if that PR gets merged, the conversion will still not be complete?

We need to track:
- [ ] https://github.com/kubernetes/kubernetes/pull/106305
- [ ] https://github.com/kubernetes/kubernetes/issues/106276
- [ ] https://github.com/kubernetes/kubernetes/issues/106262

**Comment by serathius** [comment]
Ok, only https://github.com/kubernetes/kubernetes/pull/106305 will go in 1.23 as we just need approval.
/milestone v1.24

**Comment by doshyt** [comment]
Hi all, Bug triage for 1.24 here. Wanted to check what's the expectation for this issue w.r.t 1.24? Is it still relevant and planned to be completed? Also, I see quite a few dependencies in its history - would be great to get an updated summary of blockers if any, and see whether 1.24 is still a relevant target. Thanks!

**Comment by pohly** [comment]
https://github.com/kubernetes/kubernetes/pull/108159 should finish the conversion.

**Comment by liggitt** [comment]
fixed by https://github.com/kubernetes/kubernetes/pull/108159
